File: csharp/allergies/Allergies.cs
using System;
using System.Linq;

public enum Allergen
{
    Eggs          = 1 << 0,
    Peanuts       = 1 << 1,
    Shellfish     = 1 << 2,
    Strawberries  = 1 << 3,
    Tomatoes      = 1 << 4,
    Chocolate     = 1 << 5,
    Pollen        = 1 << 6,
    Cats          = 1 << 7
}

public class Allergies
{
    private int _mask;

    public Allergies(int mask) => _mask = mask;

    public bool IsAllergicTo(Allergen allergen)
    {
      return (_mask & (int)allergen) != 0;
    }

    public Allergen[] List()
    {
      return Enum
        .GetValues(typeof(Allergen))
        .Cast<Allergen>()
        .Where(a => IsAllergicTo(a))
        .ToArray();
    }
}

File: python/allergies/allergies.py
_ALLERGENS = (
    "eggs",
    "peanuts",
    "shellfish",
    "strawberries",
    "tomatoes",
    "chocolate",
    "pollen",
    "cats",
)


def _get_allergens(score: int) -> list[str]:
    _allergens = []
    for i, allergen in enumerate(_ALLERGENS):
        bit_n = 1 << i
        if score & bit_n:
            _allergens.append(allergen)
    return _allergens


class Allergies:
    def __init__(self, score: int):
        self._allergens = _get_allergens(score)

    def allergic_to(self, item: str) -> bool:
        return item in self._allergens

    @property
    def lst(self) -> list[str]:
        return self._allergens
